### ISSUE TITLE
Implement DataController.insertContent()

### ISSUE THREAD

**Issue by Reinmar** [ckeditor/ckeditor5-engine#648]
Needed for ckeditor/ckeditor5-clipboard#1.

It seems that it must accept *view* document fragment, because it will need to convert it to the model using the context of insertion. And this becomes part of the "insert content" logic, so should most likely be done inside this method.

BTW, it made me think that this method would make sense on the `DataController`. And I kinda like this idea. We've been thinking where to move the composer and I can see that `modifySelection()` makes for the `EditingController` and `deleteContent()` together with `insertContent()` could perhaps end in the the `DataController`. Note also that we'll need something like `getSelectedContent()` for the copy and cut operations (so `deleteContent()` would become `deleteSelectedContent()`).

OTOH, I know what you're going to say – those methods look weird on the `DataController`. I agree... I'd have to get used to them. But consider this – let's say we'd like to move `insertContent()` to the `Document`. But `insertContent()` requires (as explained above) a knowledge about `DataController`. So how could it be implemented on the `Document`?

tl;dr:

* `DataController#insertContent()` – paste and I'm sure it'll be used in many ways in many cases by developers,
* `DataController#deleteSelectedContent()` – delete feature,
* `DataController#getSelectedContent()` – copy, cut and this has been also requested many times in the past for various integration scenariosm
* `EditingController#modifySelection()`

Does it make any sense or no sense at all?

**Comment by Reinmar** [comment]
There's +1 from @pjasiun and @fredck for having these method on the `DataController`. Note, however, that they will have to accept the selection on which they operate, just like now, when they are in the composer. This is fine and gives the developer the ability to specify on what piece of content the operations should be made.

**Comment by Reinmar** [comment]
PS. `modifySelection()` makes most sense in the `model.Document` or on the `Selection` as a static method.

**Comment by Reinmar** [comment]
I'll reported a two followups:

* https://github.com/ckeditor/ckeditor5-engine/issues/652
* https://github.com/ckeditor/ckeditor5-list/issues/18
